fiftyone/utils/ultralytics.py

```diff
@@ -667,3 +680,162 @@ def _to_detections(self, df):
+
+
+class UltralyticsPostProcessor(object):
+    # pylint: disable=not-callable
+    post_processor = None
+
+    def post_process(self, output):
+        imgs = output.get("imgs", None)
+        orig_imgs = output.get("orig_imgs", None)
+        if isinstance(orig_imgs, torch.Tensor):
+            orig_imgs = [i.permute(1, 2, 0).numpy() for i in orig_imgs]
+        elif isinstance(orig_imgs, list) and len(orig_imgs):
+            if isinstance(orig_imgs[0], torch.Tensor):
+                orig_imgs = [i.permute(1, 2, 0).numpy() for i in orig_imgs]
+            elif isinstance(orig_imgs[0], Image.Image):
+                orig_imgs = [np.asarray(i.convert("RGB")) for i in orig_imgs]
+
+        preds = output["preds"]
+        if self.post_processor is None:
+            raise ValueError("Ultralytics post processor is set to None")
+
+        if imgs is None or orig_imgs is None:
+            raise ValueError(
+                "Ultralytics post processor needs transformed and original images."
+            )
+
+        out = self.post_processor(preds, imgs, orig_imgs)
+        return out
+
+
+class UltralyticsClassificationOutputProcessor(
+    fout.ClassifierOutputProcessor, UltralyticsPostProcessor
+):
+    """Converts Ultralytics Classification model outputs to FiftyOne format."""
+
+    def __init__(self, classes=None, store_logits=False, post_processor=None):
+        super().__init__(classes=classes, store_logits=store_logits)
+        self.post_processor = post_processor
+
+    def __call__(self, output, frame_size, confidence_thresh=None):
+        results = self.post_process(output)
+        classifications = []
+        for result in results:
+            logits = result.probs.data.detach().cpu().numpy()
+            score = result.probs.top1conf.detach().cpu().numpy()
+            label = self.classes[result.probs.top1]
+
+            if confidence_thresh is not None and score < confidence_thresh:
+                classification = None
+            else:
+                classification = fol.Classification(
+                    label=label,
+                    confidence=score,
+                )
+                if self.store_logits:
+                    classification.logits = logits
+            classifications.append(classification)
+        return classifications
+
+
+class UltralyticsDetectionOutputProcessor(
+    fout.DetectorOutputProcessor, UltralyticsPostProcessor
+):
+    """Converts Ultralytics Detection model outputs to FiftyOne format."""
+
+    def __init__(
+        self,
+        classes=None,
+        post_processor=None,
+    ):
+        super().__init__(classes)
+        self.post_processor = post_processor
+
+    def __call__(self, output, frame_size, confidence_thresh=None):
+        results = self.post_process(output)
+        preds = self._to_dict(results)
+        return super().__call__(preds, frame_size, confidence_thresh)
+
+    def _to_dict(self, results):
+        batch = []
+        for result in results:
```

Comment:
Calling `to_detections()` here rather than reimplementing it would be consistent with the rest of the module.